src/vector.js

```diff
@@ -220,70 +222,34 @@ class Vector {
     let result = false;
 
     try {
-      let params = Vector.operate(v1, v2, [{ f: Vector.vectorSub, i: 0 }]);
+      let params = Vector.operate(v1, v2, ['Sub']);
       result = new Vector(...params);
     } catch (error) {
       console.log(`ERROR: ${error}`);
-    } finally {
-      // do nothing
     }
     return result;
   }
 
-  static mul(scalar, vectorObj) {
+  static mul(v1, scalar) {
     let result = false;
 
     try {
-      let scaleArray = [];
-      let coords;
-
-      if (is(vectorObj, 'Vector')) {
-        coords = vectorObj.coords;
-      } else if (is(vectorObj, 'Array')) {
-        coords = vectorObj;
-      }
-
-      for (let i = 0; i < coords.length; i++) {
-        scaleArray.push(scalar);
-      }
-
-      let params = Vector.operate(vectorObj, scaleArray, [
-        { f: Vector.vectorMul, i: 1 }
-      ]);
+      let params = Vector.operate(v1, scalar, ['Mul']);
       result = new Vector(...params);
     } catch (error) {
       console.log(`ERROR: ${error}`);
-    } finally {
-      // do nothing
     }
     return result;
   }
 
-  static div(scalar, vectorObj) {
```

Comment:
Nit picky but I preferred the scalar as the first argument as was previously implemented. This also introduces a problem of code not being backwards compatible (big thing to keep in mind)

Reply:
Argument order rolled back, currently pushed to this branch.